homeassistant/components/binary_sensor/isy994.py

```diff
@@ -191,12 +197,21 @@ def _positive_node_control_handler(self, event: object) -> None:
 
     # pylint: disable=unused-argument
     def on_update(self, event: object) -> None:
-        """Ignore primary node status updates.
-
-        We listen directly to the Control events on all nodes for this
-        device.
+        """Primary node status updates.
+
+        We MOSTLY ignore these updates, as we listen directly to the Control
+        events on all nodes for this device. However, there is one edge case:
+        If a leak sensor is unknown, due to a recent reboot of the ISY, the
+        status will get updated to dry upon the first heartbeat. This status
+        update is the only way that a leak sensor's status changes without
+        an accompanying Control event, so we need to watch for it.
         """
-        pass
+        if self._status_was_unknown and self._computed_state is None:
+            # pylint: disable=protected-access
+            self._computed_state = bool(self._node.status._val)
```

Comment:
Is it necessary to access the private attribute `_val` to get the value? It seems weird that the only way to access the value would by a private attribute. What does `_node.status` return?

Reply:
So, the `status` object here is odd. It's a unique [VarEvents](https://github.com/automicus/VarEvents) object that is meant to clone all of the functionality of a built-in type, but with the added ability to have code subscribe to updates to its value.

While I can access `self._node.status` and it shows up as a string '0' or '255', trying to cast it to a bool causes the code to silently halt with no exception and I cannot figure out why.

In the end I decided to go with what the base `ISYDevice` class does in its `value()` method, which was, in fact, originally implemented that way by the creator of the `VarEvents` package.

Reply:
Does `bool(int(self._node.status))` work?

Reply:
I just tested that, and that does indeed work! I pushed it up just now because I suppose it's slightly better than accessing a protected property, even though it is a bit of a hack